lib/cc/cli/init.rb

```diff
@@ -70,19 +70,21 @@ def available_configs
       end
 
       def engines_enabled?
-        unless @engines_enabled.nil?
-          return @engines_enabled
-        end
+        return @engines_enabled unless @engines_enabled.nil?
 
-        if filesystem.exist?(CODECLIMATE_YAML)
-          config = CC::Analyzer::Config.new(File.read(CODECLIMATE_YAML))
-          @engines_enabled ||= config.engine_names.any?
-        end
+        cc_config = existing_cc_config
+        @engines_enabled ||= cc_config.present? && cc_config.engine_names.any?
```

Comment:
I'm not sure how valuable this memoization is. Most of the work is probably done in `existing_cc_config` and nothing will be memoized if `present?` or `any?` return false.